### PR TITLE
Update Plc::ScriptCompletionTask to inherit from Level

### PR DESCRIPTION
**Pull request by joshlory**
Creates a new `ScriptCompletion` level type that follows the pattern of `Plc::ScriptCompletionTask`.

Eventually `LevelGroup` should own showing 2-10 inline levels as a single group of levels, but today `LevelGroup` only supports question-type levels.

### CODE REVIEW

File: dashboard/app/models/user_script.rb [hunk 1]
```diff
@@ -39,9 +39,12 @@ def empty?
...
 
-  def check_plc_task_assignment_updating
-    if self.completed_at && self.script.pd
-      Plc::ScriptCompletionTask.check_for_script_completion(self)
+  def mark_script_completion_levels
+    if completed_at && script.pd
+      ScriptCompletion.all.
+        select{ |sc| sc.script_id == script.id.to_s }.
+        map(&:script_levels).flatten.
+        each{ |sl| user.track_level_progress_async(sl, ActivityConstants::BEST_PASS_RESULT, false) }
```
Comment (mehalshah) [inline dashboard/app/models/user_script.rb:47]: prefer each do |sl|
...
end

File: dashboard/test/models/user_script_test.rb [hunk 2]
```diff
@@ -75,42 +77,21 @@ class UserScriptTest < ActiveSupport::TestCase
...
+    level.script_id = @script.id.to_s
+    level.save!
+    sl = create :script_level, levels: [level]
+
+    refute UserLevel.exists?(user: @user, level: sl.level, script: sl.script)
+    complete_all_levels
+    User.track_script_progress(@user, @script)
+    assert UserLevel.find_by(user: @user, level: sl.level, script: sl.script).perfect?
   end
 end
```
Comment (mehalshah) [inline dashboard/test/models/user_script_test.rb:117]: Add tests for
- Non PD scripts should not look for scripts to complete (basically, prove that this code won't get executed a bunch during HOC)
- Completing some levels does not mark the userlevel as complete

File: dashboard/app/models/user_script.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/app/models/user_script.rb:47]: Our style guide recommends `{...}` over `do...end` for chaining: https://github.com/bbatsov/ruby-style-guide#single-line-blocks

(It also recommends not chaining across multiple lines, but [we tend to do this in our codebase](https://github.com/code-dot-org/code-dot-org/blob/24343fad6bf1327fda31714ea13da766587d0a5e/dashboard/app/models/plc/script_completion_task.rb#L31-L35).)

File: dashboard/test/models/user_script_test.rb (hunk 2 above)
Comment (joshlory) [inline dashboard/test/models/user_script_test.rb:117]: Added in f5f5cb6d5a030a344ad5d6eec726e112080edc85.

File: dashboard/test/models/user_script_test.rb [hunk 3]
```diff
@@ -75,42 +75,43 @@ class UserScriptTest < ActiveSupport::TestCase
...
+
+    # Complete remaining levels
+    @script_levels[8..-1].each do |script_level|
+      complete_level(script_level)
+    end
+
+    assert UserLevel.find_by(user: @user, level: sl.level, script: sl.script).perfect?
+  end
+
+  test "completing a pd script scans ScriptCompletion" do
```
Comment (joshlory) [inline dashboard/test/models/user_script_test.rb:105]: I'm not sure this is needed, but including for completeness.  The method could change so `ScriptCompletion.all` is never called, and the next test would still pass.

**Comment by mehalshah** [comment]
LGTM

**Pull request merged**